### DISCUSSION TITLE
Interpolation function

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#26370] [Q&A General]
Hello,
PiecewiseMultilinear function provides the capability of multi-dimensional piecewise linear interpolation depending only on spatial position and time. But I want to interpolate other things like variables, so does MOOSE provide an interface or function to do this?

**Comment by lindsayad** [comment]
Not in the same gridded data way. But you can use a [ParsedMaterial](https://mooseframework.inl.gov/source/materials/ParsedMaterial.html) to define material properties that are arbitrary expressions of space, time, coupled variables, postprocessors, and other material properties

**Comment by GiudGiud** [comment]
The underlying utilities that are leveraged by PiecewiseLinear interpolation functions can also be used in a new object with variables, etc as the operands (or even the grid) of the interpolation. 

you will have to code a new object though
